extensions/indexes/range/src/org/exist/indexing/range/RangeIndexWorker.java

```diff
@@ -512,7 +512,7 @@ private void write() {
     public NodeSet query(int contextId, DocumentSet docs, NodeSet contextSet, List<QName> qnames, AtomicValue[] keys, RangeIndex.Operator operator, int axis) throws IOException, XPathException {
         return index.withSearcher(searcher -> {
             List<QName> definedIndexes = getDefinedIndexes(qnames);
-            NodeSet resultSet = NodeSet.EMPTY_SET;
+            NodeSet resultSet = new NewArrayNodeSet();
```

Comment:
I don't think we need to check for empty here. Adding an empty sequence to an empty sequence should result in an empty sequence ;-)